test/inductor/test_torchinductor.py

```diff
@@ -7969,6 +7970,88 @@ def fn(x):
 
         self.common(fn, (torch.randn((16, 32)),), check_lowp=False)
 
+    @requires_cuda()
+    @torch._inductor.config.patch("layout_optimization", True)
+    @torch._inductor.config.patch("keep_output_stride", False)
+    @config.patch(implicit_fallbacks=True)
+    def test_custom_op_fixed_layout(self):
+        import torch.library
+
+        mod = nn.Conv2d(3, 128, 1, stride=1, bias=False).cuda()
+        inp = torch.rand(2, 3, 128, 128, device="cuda")
+        expected_stride = mod(inp).stride()
+
+        def foo_cpu(x):
+            self.assertEqual(x.stride(), expected_stride)
+            return x.clone()
+
+        def foo_cuda(x):
+            self.assertEqual(x.stride(), expected_stride)
+            return x.clone()
+
+        def foo_meta(x):
+            return torch.empty_like(x)
+
+        global libbar
```

Comment:
Not sure if there's a better way to do this then individual globals for each test case. I tried putting the torch.library.Library() calls into a helper function that took in references to the implementations, but then somehow the custom ops would not stay in scope.